### PR TITLE
Part four: Deep link

### PR DESCRIPTION
**Pull request by ltxitai**
Allow global deep-link for URLs of the form baygur://query. The top
result for the given query is displayed.

### CODE REVIEW

File: Baygur/BGRPixabayImageMetadataFetcher.mm [hunk 1]
```diff
@@ -79,19 +88,33 @@ - (void)fetchEditorsChoices:(NSUInteger)maxCount
   }] resume];
 }
 
-/// Creates the URL request for the given count. The Pixabay API sets a minimum of 3 items per page,
-/// so if less are requested - pass a request for 3, but only return as many as were requested.
 - (NSURL *)urlForCount:(NSUInteger)count {
+  auto components = [self baseComponentsForCount:count];
+  components.queryItems = [components.queryItems
+                           arrayByAddingObject:[NSURLQueryItem queryItemWithName:@"editors_choice"
```
Comment (gershonho) [inline Baygur/BGRPixabayImageMetadataFetcher.mm]: 1. Separate into 2 statements - 1)create query item; 2)add it to the array.
2. Create a `mutableCopy` of `components.queryItems` and add the new item to it.

File: Baygur/BGRPixabayImageMetadataFetcher.mm [hunk 2]
```diff
@@ -79,19 +88,33 @@ - (void)fetchEditorsChoices:(NSUInteger)maxCount
...
+                                                                           value:@"1"]];
+  return components.URL;
+}
+
+- (NSURL *)urlForQuery:(NSString *)query maxCount:(NSUInteger)maxCount {
+  auto components = [self baseComponentsForCount:maxCount];
+  components.queryItems = [components.queryItems
+                           arrayByAddingObject:[NSURLQueryItem queryItemWithName:@"q" value:query]];
```
Comment (gershonho) [inline Baygur/BGRPixabayImageMetadataFetcher.mm]: Same

File: Baygur/BGRPixabayImageMetadataFetcher.mm (hunk 1 above)
Comment (ltxitai) [reply Baygur/BGRPixabayImageMetadataFetcher.mm]: Done.

File: Baygur/BGRPixabayImageMetadataFetcher.mm (hunk 2 above)
Comment (ltxitai) [reply Baygur/BGRPixabayImageMetadataFetcher.mm]: Done.

**Pull request merged**